tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1481,6 +1504,61 @@ public:
    size_t GetAlignment() const final { return std::alignment_of<ContainerT>(); }
 };
 
+template <typename T1, typename T2>
+class RField<std::pair<T1, T2>> : public RPairField {
+private:
+   using ContainerT = typename std::pair<T1,T2>;
+   template <typename Ty1, typename Ty2>
+   static std::vector<std::unique_ptr<Detail::RFieldBase>> BuildItemFields()
+   {
+      std::vector<std::unique_ptr<Detail::RFieldBase>> result;
+      result.emplace_back(new RField<Ty1>("_0"));
+      result.emplace_back(new RField<Ty2>("_1"));
```

Comment:
We use generally the `_0`, `_1`, ... naming for the subfields created in the context of STL types. That makes it easier to read them back as another type (e.g., record as pair, RVec as std::vector, etc). So I think for the on-disk storage, the `_0` and `_1` names are all right.

At some point, I think we should add some name resolution logic that gives users more flexibility in accessing fields. In unambiguous cases, for instance, users should be able to access `std::pair` subfields also as `first` and `second`.

Reply:
> That makes it easier to read them back as another type (e.g., record as pair, RVec as std::vector, etc).

I am confused.  What do you refer to as sub-field of std::vector?  I can see only 2 scenarios, a `vector` of numerical type where indeed a generic name make sense or a `vector` of struct where (unless you have that information somewhere else) it is essential for schema evolution (eg. data members re-ordered) to associate the name the members with the field.  (There is a 3rd case for a `map` where anonymous field name might be okay or using `first` and `second` might help convert from `map` to `vector of pair`).

Reply:
For `vector<T>` , there is always a subfield named `_0`, independently of `T`. This subfield can have further subfields if `T` is a complex type. The type name (and type version) of `T` is stored in an extra string on disk, i.e. there is a field name and a field type (+ type version).

Reply:
I see.  For collection, I can see how it makes sense and the `_0` is then the `generic` name for the implementation dependent data member name that hold the actually array of objects (per se) [Compared to the way, `TTree` does it, it has the disadvantage of being 'verbose-ish' to refer to (`myvector._0.datamember` vs `myvector.datamember` but it has the advantage of giving access to the `vector` itself without `TTreeFormula`'s weirdness (`myvector.@size()`).

However, I still don't understand why this would be the case for `std::pair`, I don't see any good reason why the naming of pair should differ to the one obtains from
```
class MyPair {
   int first;
   double second;
};
or
template <typename T1, typename T2> MyGenericPair {
   T1 first;
   T2 second;
};
```
and having "what we can do in this PR is patching the RDF data source" is clearing an indication of the problem (i.e. another tools that is similar to RDF would have to have the same hack/patch where-as if `std::pair` was treated as anyother struct, then it's support would come for free (from supporting struct).

Reply:
I'm tend to agree with @jblomer's opinion; it is more aligned with the naming of subfields for other types (e.g. `std::variant<Ts...>` or `std::vector<T>`.

IMO, what we can do, however, is to provide `first` and `second` as aliased columns in RNTuple (not RDF), once that is supported.